pkgs/development/compilers/nextpnr/default.nix

```diff
@@ -1,11 +1,7 @@
 { stdenv, fetchFromGitHub, cmake, makeWrapper
-, boost, python3
+, boost, python3, eigen
 , icestorm, trellis
-
-# TODO(thoughtpolice) Currently the GUI build seems broken at runtime on my
-# laptop (and over a remote X server on my server...), so mark it broken for
-# now, with intent to fix later.
-, enableGui ? false
```

Comment:
I'd prefer if this was left in. I should have updated the comment, but: the reason this is still gated is because the NextPNR GUI is broken with QT 5.12, but it is *not* broken with QT 5.11. I do not know the reasons for this.

Now, normally, that wouldn't be a problem, after all, Nix is supposed to handle this. But the thing is, the only reason we even *have* qt511 is for Darwin support -- Qt 5.12 is broken there for now. On Linux, qt512 is not only the default, but there are *no* other packages other than nextpnr which require qt511.

So the reason I left it out is: it's an odd duck that is the only Linux package that needs an older QT. In the event we move to qt 5.12 (e.g. it's fixed on Darwin, or 5.13 is fixed and we move there), I would prefer not to hold up everything on nextpnr, which is a non-critical package. But at the same time, just bumping it breaks the UX, which is really unfortunate. I wasn't sure what time-table we were on for removing/updating QT versions either, so in lieu of debugging things, I left this as is.

This is a long way of saying: I think we should fix the GUI on 5.12, and *only then* enable it by default.

Reply:
For the record, the GUI is broken in the sense of "it's got some missing UI elements", not "it doesn't run and is completely unusable". I mean, it *is* unusable if you want to look at the floorplan, but it doesn't crash.

IIRC, the problem is the black background of the floorplan UX (where the wires are drawn) does not actually get drawn in -- this results in the black background becoming *fully transparent* and allowing you to "see through" it like a window pane. But the wires *do* still get drawn. And, because you normally, you know, have things *behind* a window (like your desktop wallpaper), the floorplan UX becomes unusable because you cannot see 3px wide wires on top of your background.

I assume this is perhaps something to do with a missing Qt module, theme, shader, something or another that means the Qt application can't find the resource to "bind" to that background, so it fails. Presumably someone with more Qt experience can figure it out...

Reply:
That makes sense; I put it back.